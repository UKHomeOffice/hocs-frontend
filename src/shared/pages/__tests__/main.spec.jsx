import React from 'react';
import Main from '../main.jsx';
import { MemoryRouter } from 'react-router-dom';

jest.mock('../../common/components/workstack.jsx', () => (
    'MOCK_WORKSTACK'
));

describe('Main page component', () => {

    const mockDispatch = jest.fn();

    beforeEach(() => {
        mockDispatch.mockReset();
    });

    it('should render with default props', () => {
        const wrapper = mount(
            <MemoryRouter>
                <Main />
            </MemoryRouter>
        );
        expect(wrapper).toBeDefined();
        expect(wrapper.find('MainPage')).toMatchSnapshot();
    });

    it('should render with caption when passed in props', () => {
        const wrapper = mount(
            <MemoryRouter>
                <Main caption='MY CAPTION' />
            </MemoryRouter>
        );
        expect(wrapper).toBeDefined();
        expect(wrapper.find('MainPage')).toMatchSnapshot();
    });

});