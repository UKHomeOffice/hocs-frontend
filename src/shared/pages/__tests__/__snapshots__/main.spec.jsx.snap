// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main page component should render with caption when passed in props 1`] = `
<MainPage
  caption="MY CAPTION"
>
  <div
    className="govuk-grid-row"
  >
    <div
      className="govuk-grid-column-full"
    >
      <h1
        className="govuk-heading-l"
      >
        <span
          className="govuk-caption-l"
        >
          MY CAPTION
        </span>
      </h1>
      <h2
        className="govuk-heading-m"
      >
        Primary actions
      </h2>
      <ul
        className="govuk-list govuk-list--bullet"
      >
        <li>
          <Link
            replace={false}
            to="/action/create/workflow"
          >
            <a
              href="/action/create/workflow"
              onClick={[Function]}
            >
              Create single case
            </a>
          </Link>
        </li>
        <li>
          <Link
            replace={false}
            to="/action/bulk/workflow"
          >
            <a
              href="/action/bulk/workflow"
              onClick={[Function]}
            >
              Create cases in bulk
            </a>
          </Link>
        </li>
        <li>
          <Link
            replace={false}
            to="/action/test/form"
          >
            <a
              href="/action/test/form"
              onClick={[Function]}
            >
              View test form
            </a>
          </Link>
        </li>
      </ul>
      <h2
        className="govuk-heading-m"
      >
        Secondary routes
      </h2>
      <ul
        className="govuk-list govuk-list--bullet"
      >
        <li>
          <Link
            replace={false}
            to="/action/random/url"
          >
            <a
              href="/action/random/url"
              onClick={[Function]}
            >
              Test 404
            </a>
          </Link>
        </li>
        <li>
          <Link
            replace={false}
            to="/action/test/form"
          >
            <a
              href="/action/test/form"
              onClick={[Function]}
            >
              Test 403
            </a>
          </Link>
        </li>
        <li>
          <Link
            replace={false}
            to="/error"
          >
            <a
              href="/error"
              onClick={[Function]}
            >
              Test 500
            </a>
          </Link>
        </li>
      </ul>
      <MOCK_WORKSTACK />
    </div>
  </div>
</MainPage>
`;

exports[`Main page component should render with default props 1`] = `
<MainPage>
  <div
    className="govuk-grid-row"
  >
    <div
      className="govuk-grid-column-full"
    >
      <h1
        className="govuk-heading-l"
      />
      <h2
        className="govuk-heading-m"
      >
        Primary actions
      </h2>
      <ul
        className="govuk-list govuk-list--bullet"
      >
        <li>
          <Link
            replace={false}
            to="/action/create/workflow"
          >
            <a
              href="/action/create/workflow"
              onClick={[Function]}
            >
              Create single case
            </a>
          </Link>
        </li>
        <li>
          <Link
            replace={false}
            to="/action/bulk/workflow"
          >
            <a
              href="/action/bulk/workflow"
              onClick={[Function]}
            >
              Create cases in bulk
            </a>
          </Link>
        </li>
        <li>
          <Link
            replace={false}
            to="/action/test/form"
          >
            <a
              href="/action/test/form"
              onClick={[Function]}
            >
              View test form
            </a>
          </Link>
        </li>
      </ul>
      <h2
        className="govuk-heading-m"
      >
        Secondary routes
      </h2>
      <ul
        className="govuk-list govuk-list--bullet"
      >
        <li>
          <Link
            replace={false}
            to="/action/random/url"
          >
            <a
              href="/action/random/url"
              onClick={[Function]}
            >
              Test 404
            </a>
          </Link>
        </li>
        <li>
          <Link
            replace={false}
            to="/action/test/form"
          >
            <a
              href="/action/test/form"
              onClick={[Function]}
            >
              Test 403
            </a>
          </Link>
        </li>
        <li>
          <Link
            replace={false}
            to="/error"
          >
            <a
              href="/error"
              onClick={[Function]}
            >
              Test 500
            </a>
          </Link>
        </li>
      </ul>
      <MOCK_WORKSTACK />
    </div>
  </div>
</MainPage>
`;
